src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
+     *
+     * @return StreamInterface
```

Comment:
Hm, actually creating multiple factories are a bit hard IMO. Since at the moment both createStream and createMultipartStream must be implemented, I would expect having one factory for Guzzle, one for Diactoros, etc.